tests/utils/exec.ts

```diff
@@ -629,171 +613,29 @@ async function setupTestAgent(privateKeyPem: PrivateKeyPem, logger: Logger) {
-    child_process.execFile(
-      'nsenter',
-      [
-        ...nsenter(usrnsPid, netnsPid),
-        'ts-node',
-        '--project',
-        tsConfigPath,
-        polykeyPath,
-        ...args,
-      ],
-      {
-        env,
-        cwd,
-        windowsHide: true,
-      },
-      (error, stdout, stderr) => {
-        if (error != null && error.code === undefined) {
-          // This can only happen when the command is killed
-          return reject(error);
-        } else {
-          // Success and Unsuccessful exits are valid here
-          return resolve({
-            exitCode: error && error.code != null ? error.code : 0,
-            stdout,
-            stderr,
-          });
-        }
-      },
-    );
-  });
-}
-
-/**
- * Launch pk command through subprocess inside a network namespace
- * This is used when a subprocess functionality needs to be used
- * This is intended for non-terminating subprocesses
- * @param env Augments env for command execution
- * @param cwd Defaults to temporary directory
- */
-async function pkSpawnNs(
-  usrnsPid: number,
-  netnsPid: number,
-  args: Array<string> = [],
-  env: Record<string, string | undefined> = {},
-  cwd?: string,
-  logger: Logger = new Logger(pkSpawnNs.name),
-): Promise<ChildProcess> {
-  cwd =
-    cwd ?? (await fs.promises.mkdtemp(path.join(os.tmpdir(), 'polykey-test-')));
-  env = {
-    ...process.env,
-    ...env,
-  };
-  // Recall that we attempt to connect to all specified seed nodes on agent start.
-  // Therefore, for testing purposes only, we default the seed nodes as empty
-  // (if not defined in the env) to ensure no attempted connections. A regular
-  // PolykeyAgent is expected to initially connect to the mainnet seed nodes
-  env['PK_SEED_NODES'] = env['PK_SEED_NODES'] ?? '';
-  const tsConfigPath = path.resolve(
-    path.join(globalThis.projectDir, 'tsconfig.json'),
-  );
-  const polykeyPath = path.resolve(
-    path.join(globalThis.projectDir, 'src/bin/polykey.ts'),
-  );
-  const subprocess = child_process.spawn(
-    'nsenter',
-    [
-      ...nsenter(usrnsPid, netnsPid),
-      'ts-node',
-      '--project',
-      tsConfigPath,
-      polykeyPath,
-      ...args,
-    ],
-    {
-      env,
-      cwd,
-      stdio: ['pipe', 'pipe', 'pipe'],
-      windowsHide: true,
-      shell: true,
-    },
-  );
-  const rlErr = readline.createInterface(subprocess.stderr!);
-  rlErr.on('line', (l) => {
-    // The readline library will trim newlines
-    logger.info(l);
-  });
-  return subprocess;
+function escapeShellArgs(arg: string): string {
+  return arg.replace(/(["\s'$`\\])/g, '\\$1');
 }
 
 export {
-  exec,
-  pk,
-  pkStdio,
+  polykeyPath,
   pkExec,
   pkSpawn,
-  pkStdioTarget,
-  pkExecTarget,
-  pkSpawnTarget,
   pkExpect,
+  pkStdio,
+  exec,
+  spawn,
+  pk,
+  setupTestAgent,
   processExit,
   expectProcessError,
-  setupTestAgent,
+  pkExecWithShell,
+  pkExecWithoutShell,
+  pkSpawnWithShell,
+  pkSpawnWithoutShell,
   spawnFile,
```

Comment:
Can this be replaced with `testUtils.spawn`?

Reply:
Not without some changes. The API for `testUtils.spawn` is slightly different since it's not meant to be used for non-polykey processes. Those four aren't used anywhere at the moment, they're just there for edge cases.

Reply:
> The API for testUtils.spawn is slightly different since it's not meant to be used for non-polykey processes.

I thought `testUtils.spawn` and `testUtils.exec` are intended for non-polykey processes?

Reply:
Oh sorry I thought you were referring to the `pkExec...`/`pkSpawn...` variants. I'm not really sure what `spawnFile` does. It's only used in `NodeConnection.test.ts`, but I didn't write those tests.

Reply:
Have a look at `NodeConnection.test.ts`. It's just acquiring the test process as a dummy process for GRPC. If your `testUtils.spawn` is simple enough, then it should work fine. You can swap it over and remove `spawnFile`.